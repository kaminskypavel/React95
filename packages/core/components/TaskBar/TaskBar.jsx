import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Button from "../Button";
import Icon from "../Icon";

const Wrapper = styled.div`
  flex: 1 0 auto;
  height: 800px;
  flex-direction: revert;
`

const Bar = styled.footer`
  padding: 3px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff`;


const TaskBar = ({onStart}) => {
    return (
      <Wrapper>
        <Bar>
          <Button onClick={onStart} style={{display: 'flex', flexDirection: 'row'}}>
            <Icon style={{alignSelf: 'center', maxHeight: 20,maxWidth: 20, padding: 0,margin:0,flex:1}} name="logo" />
            <b style={{alignSelf: 'center', paddingLeft: 5}}>Start</b>
          </Button>
        </Bar>
      </Wrapper>
    );
};

TaskBar.displayName = 'TaskBar';

TaskBar.propTypes = {
    onStart: PropTypes.func,
};

TaskBar.defaultProps = {
    onStart: () => {
    },
};

export default TaskBar;
