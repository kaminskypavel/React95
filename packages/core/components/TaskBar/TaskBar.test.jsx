import React from 'react';
import { render } from '@testing-library/react';
import TaskBar from './TaskBar';

describe('<TaskBar />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with error type', () => {
      const {container} = render(<TaskBar />);
      expect(container).toMatchSnapshot();
    });
});
});
