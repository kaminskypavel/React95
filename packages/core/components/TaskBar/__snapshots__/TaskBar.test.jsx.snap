// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskBar /> Snapshots should match snapshot with error type 1`] = `
.c2 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 0px #000;
}

.c2:disabled {
  color: #868a8e;
}

.c2:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -0.5px -0.5px 0px 1px #868a8e,1px 1px 0 1px #000;
  outline: 1px dotted #000;
  outline-offset: -5px;
}

.c2:active {
  padding: 8px 20px 5px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

.c3 {
  display: block;
  width: 32px;
  height: 32px;
  background-image: url('test-file-stub');
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}

.c0 {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  height: 800px;
  -webkit-flex-direction: revert;
  -ms-flex-direction: revert;
  flex-direction: revert;
}

.c1 {
  padding: 3px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff;
}

<div>
  <div
    class="c0"
  >
    <footer
      class="c1"
    >
      <button
        class="c2"
        style="display: flex; flex-direction: row;"
        value="Ok"
      >
        <i
          class="c3"
          height="32"
          name="logo"
          style="align-self: center; max-height: 20px; max-width: 20px; padding: 0px; margin: 0px; flex: 1;"
          width="32"
        />
        <b
          style="align-self: center; padding-left: 5px;"
        >
          Start
        </b>
      </button>
    </footer>
  </div>
</div>
`;
